src/main/java/uk/gov/companieshouse/auth/filter/AuthFilter.java

```
@@ -0,0 +1,209 @@
+     * @param originalRequestUri Original URI from which to redirect
+     * @param scope Scope of the request
+     * @param nonce
+     * @param email Email address from the session, empty if not present
+     * @return Authorisation URI
+     */
+    protected String createAuthoriseURIWithForceAndHint(String originalRequestUri, String scope, String nonce, String email) {
+        
+        StringBuilder sb = new StringBuilder(createAuthoriseURI(originalRequestUri, scope, nonce));      
+        sb.append("&reauthenticate=force");
+        sb.append("&hint=");
+        sb.append(jweEncodeWithNonce(email, nonce, "email"));
+        
+        return sb.toString(); 
+    }
+
+    /**
+     * Generates a secure unique key
+     * @return Base64 encoded unique key
+     */
+    private String generateNonce() {
+    
+        SecureRandom random = new SecureRandom();
+        byte[] bytes = new byte[5];
+        random.nextBytes(bytes);
+        
+        return Base64.encodeBase64URLSafeString(bytes);        
+    }
+
+    /**
+     * Encodes a URI with a nonce according to a JWE encoding algorithm
+     * @param returnUri
+     * @param nonce
+     * @param attributeName
+     * @return JWE encoded string, comprised of the return URI and a nonce
+     */
+    protected String jweEncodeWithNonce(String returnUri, String nonce, String attributeName) {
+        
+        JSONObject payloadJson = new JSONObject();
+        payloadJson.put(attributeName, returnUri);
+        payloadJson.put("nonce", nonce);
+        
+        Payload payload = new Payload(payloadJson);
+        JWEHeader header = new JWEHeader(JWEAlgorithm.DIR, EncryptionMethod.A128CBC_HS256);
+        JWEObject jweObject = new JWEObject(header, payload);
+
+        byte[] key = Base64.decodeBase64(base64Key);
+        
+        try {
+            jweObject.encrypt(new DirectEncrypter(key));
+        } catch (JOSEException e) {
+            log.error(e, null);
```

My understanding of the code: If encryption fails then we catch the exception, log it and serialise a non-encrypted JWE object. Is this correct behaviour?

---

The call to serialize will throw an `IllegalStateException` if the jweObject is not encrypted. Is that sufficient, or should we return a blank string inside the catch?

---

Why aren't you throwing the encryption exception? You are swallowing an exception then throwing a different one caused by the first exception.